packages/lib-classifier/src/store/Classification.js

```diff
@@ -46,6 +47,43 @@ const Classification = types
     }),
     metadata: types.maybe(ClassificationMetadata)
   })
+  .views(self => ({
+    get toSnapshot () {
+      let snapshot = getSnapshot(self)
+      let annotations = []
+      self.annotations.forEach(annotation => {
+        annotations = annotations.concat(annotation.toSnapshot)
+      })
+      snapshot = Object.assign({}, snapshot, { annotations })
+      console.log('Panoptes classification', snapshot)
+      return snapshot
+    }
+  }))
+  .preProcessSnapshot(snapshot => {
+    console.log('Classification preprocessor', snapshot)
+    const newSnapshot = Object.assign({}, snapshot)
+    // generate classification IDs, if not present
+    newSnapshot.id = snapshot.id || cuid()
+    // convert any annotations arrays to a map
+    const annotationsMap = {}
+    if (snapshot.annotations && Array.isArray(snapshot.annotations)) {
+      snapshot.annotations.forEach(annotation => annotationsMap[annotation.task] = annotation)
+      newSnapshot.annotations = annotationsMap
+    }
+    console.log(newSnapshot)
+    return newSnapshot
+  })
+  .postProcessSnapshot(snapshot => {
+    console.log('Classification postprocessor')
+    const newSnapshot = Object.assign({}, snapshot)
+    // remove temporary classification IDs
+    // TODO: leave the ID if it came from Panoptes
+    delete newSnapshot.id
+    // convert annotations to an array
+    newSnapshot.annotations = Object.values(snapshot.annotations)
```

Comment:
Why not use the `convertMapToArray` helper already written and tested?

Reply:
Does that work on plain objects? I'm not sure if the snapshot of an observable map is a JS Map.

Reply:
I just tried it and it throws `TypeError: map.values is not a function`.

Reply:
Ok, so this sort of answers my question and is one of the concerns I have about using snapshots. We have no choice in how the observable map is converted into javascript with the snapshot, whereas `toJS` gives us that choice. The snapshot converts the observable map to just an object. We had a long discussion about how to manage the conversion, which resulted in ADR 10: https://github.com/zooniverse/front-end-monorepo/blob/master/docs/arch/adr-10.md

Reply:
Is there a plan for stripping underscored values. For instance, there's an id submitted with drawing annotations that would have no use downstream, so this should be underscored and removed:

![Screen Shot 2020-02-05 at 3 13 04 PM](https://user-images.githubusercontent.com/5016731/73883706-0781e100-482a-11ea-9fa6-fc8ccdad3591.png)

Reply:
IDs are stripped by the post-processors. See classifications and marks for examples. I'll add that to annotations too.

You can double-check yourself, but `getSnapshot` should preserve the order of maps. I'll add tests to verify that annotation order is not changed.

Reply:
I'm not seeing those IDs now that I've fixed the annotation post-processor.
![A completed classification with annotation IDs stripped from drawing and question task annotations.](https://user-images.githubusercontent.com/59547/73935676-fc39be80-48d8-11ea-8494-ed631db9d86c.png)